src/redux/reducers/searchText.js

```diff
@@ -0,0 +1,198 @@
+import { handleActions } from 'redux-actions';
+import actionTypes from '../actionTypes';
+import { searchMessages } from '../../utils/searchMessages';
+
+const defaultState = {
+  search: false,
+  searchState: {
+    markedMessagesByID: {},
```

Comment:
This state would not be required now. We will already have message by ids object in our store

Reply:
@akshatnitd True, but sir this is being used to maintain a copy of the message object that is being modified by the **SearchMessage** util

```
const orignalMessageText = messagesByID[id].text;
    let messageCopy = { ...messagesByID[id] };
    let messageText = orignalMessageText;
    if (orignalMessageText) {
      if (!isCaseSensitive) {
        searchText = searchText.toLowerCase();
        messageText = messageText.toLowerCase();
      }
      const match = messageText.indexOf(searchText);
      if (match !== -1) {
        messageCopy.mark = {
          searchText,
          isCaseSensitive,
        };
        markingData.markedIDs = [messageCopy.id, ...markingData.markedIDs];
        markingData.markedIndices = [index, ...markingData.markedIndices];
      }
      markingData.markedMessagesByID[id] = messageCopy;
```

Reply:
Any suggestions, sir? @akshatnitd